plugin/src/withBLE.ts

```diff
@@ -0,0 +1,51 @@
+import { AndroidConfig, ConfigPlugin, createRunOncePlugin, WarningAggregator } from '@expo/config-plugins'
+
+import { withBLEAndroidManifest } from './withBLEAndroidManifest'
+import { BackgroundMode, withBLEBackgroundModes } from './withBLEBackgroundModes'
+import { withBluetoothPermissions } from './withBluetoothPermissions'
+
+const pkg = { name: 'react-native-ble-plx', version: 'UNVERSIONED' } // require('react-native-ble-plx/package.json')
+
+/**
+ * Apply BLE native configuration.
+ */
+const withBLE: ConfigPlugin<
+  {
+    isBackgroundEnabled?: boolean
+    neverForLocation?: boolean
+    modes?: BackgroundMode[]
+    bluetoothAlwaysPermission?: string | false
+  } | void
+> = (config, props = {}) => {
+  const _props = props || {}
+  const isBackgroundEnabled = _props.isBackgroundEnabled ?? false
+  const neverForLocation = _props.neverForLocation ?? false
+
+  if ('bluetoothPeripheralPermission' in _props) {
+    WarningAggregator.addWarningIOS(
+      'bluetoothPeripheralPermission',
+      `The iOS permission \`NSBluetoothPeripheralUsageDescription\` is fully deprecated as of iOS 13 (lowest iOS version in Expo SDK 47+). Remove the \`bluetoothPeripheralPermission\` property from the \`@config-plugins/react-native-ble-plx\` config plugin.`
+    )
+  }
+
+  // iOS
+  config = withBluetoothPermissions(config, _props)
+  config = withBLEBackgroundModes(config, _props.modes || [])
+
+  // Android
+  config = AndroidConfig.Permissions.withPermissions(config, [
+    'android.permission.BLUETOOTH',
+    'android.permission.BLUETOOTH_ADMIN',
+    'android.permission.BLUETOOTH_CONNECT' // since Android SDK 31
```

Comment:
what about BLUETOOTH_SCAN (https://developer.android.com/reference/android/Manifest.permission#BLUETOOTH_SCAN)

Reply:
`BLUETOOTH_SCAN` is dynamically added in `addScanPermissionToManifest`